server/bin/pbench-server-put-shim.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+# Forward tar balls received via SSH from version 002 clients
+# to the configured pbench server using the pbench agent's new
+# PUT API interface.
+
+import os
+import sys
+import tempfile
+
+from pbench import PbenchConfig, BadConfig, get_pbench_logger
+from pbench.process_tb import ProcessTb
+from pbench.report import Report
+
+
+_NAME_ = "pbench-server-put-shim"
+
+
+def main(cfg_name: str) -> int:
+    if not cfg_name:
+        print(
+            f"{_NAME_}: ERROR: No config file specified; set"
+            " _PBENCH_SERVER_CONFIG env variable or use --config <file> on the"
+            " command line",
+            file=sys.stderr,
+        )
+        return 2
+
+    try:
+        config = PbenchConfig(cfg_name)
+    except BadConfig as e:
+        print(f"{_NAME_}: {e} (config file {cfg_name})", file=sys.stderr)
+        return 1
+
+    logger = get_pbench_logger(_NAME_, config)
+
+    try:
+        ptb = ProcessTb(config, logger)
+    except Exception as e:
+        logger.error("Error: {}", e)
+        return 2
+
+    counts = ptb.process_tb()
+
+    result_string = (
+        f"{config.TS} Status: Total no. of tarballs {counts.ntotal},"
+        f" Successfully moved {counts.ntbs}, Encountered"
+        f" {counts.ntotal-counts.ntbs} failures"
+    )
+    logger.info(result_string)
+
+    # prepare and send report
+    fd, tmp_file = tempfile.mkstemp(dir=config.TMP, text=True)
+    with os.fdopen(fd, "w+") as reportfp:
+        reportfp.write(f"{counts.nstatus}{result_string}\n")
+
+    report = Report(config, _NAME_)
+    report.init_report_template()
+    try:
```

Comment:
Consider moving the `try` to cover the `os.fdopen()` call.

The `tempfile.mkstemp()` call should probably have its own try/except to catch any exceptions and log them as errors.

Reply:
Just as an aside, the `fdopen()` doesn't actually open anything -- that was done by `mkstemp()` -- it's just wrapping the existing file descriptor in a Python file object.